src/DynamoCore/Library/FunctionDescriptor.cs

```diff
@@ -568,4 +569,113 @@ private string GetRootCategory()
             return string.IsNullOrEmpty(Namespace) ? filename : filename + "." + Namespace;
         }
     }
+
+    internal class ReflectionFunctionDescriptor : FunctionDescriptor
```

Comment:
deserves a summary I think.
Does it make more sense to put this in the reflection namespace instead of here?

Reply:
Which Reflection namespace are you referring to, the one in `ProtoCore`?

Reply:
yes, I was thinking in general it might make more sense to move these reflection specific classes to the protocore.reflection namespace - which I suggested get renamed to ProtoFFI.reflection ... 

not sure what you think of that reorg.

I'm just thinking that for the most part, devs who come upon this file won't usually need to be aware of this class.

Reply:
> I'm just thinking that for the most part, devs who come upon this file won't usually need to be aware of this class.

Yea that makes sense, i guess the issue is that we would need a reference to DynamoCore in ProtoCore, to inherit from the `FunctionDescriptor, which we cant as Core already references ProtoCore